cdap-elastic/src/main/java/co/cask/cdap/metadata/elastic/ElasticsearchMetadataStorage.java

```diff
@@ -319,10 +320,27 @@ public MetadataChange apply(MetadataMutation mutation) throws IOException {
 
   @Override
   public List<MetadataChange> batch(List<? extends MetadataMutation> mutations) throws IOException {
+    if (mutations.isEmpty()) {
+      return Collections.emptyList();
+    }
+    Set<MetadataEntity> entities = new HashSet<>();
```

Comment:
Is it possible to use a Map< MetadataEntity, List<MetadataMutation>> to group together the mutations for each entity, and then execute the disjoint mutations concurrently?

Reply:
yes, it is but I did not think it is worth the complexity. Do you think it should be done? I opened a Jira here: https://issues.cask.co/browse/CDAP-14979

Reply:
👍